prov/tcp/src/tcpx_ep.c

```diff
@@ -651,6 +651,54 @@ static int tcpx_pep_listen(struct fid_pep *pep)
 
 }
 
+static int tcpx_pep_reject(struct fid_pep *pep, fid_t handle,
+		    const void *param, size_t paramlen)
+{
+	struct poll_fd_info *fd_info;
+	struct tcpx_conn_handle *tcpx_handle;
+	struct tcpx_fabric *tcpx_fabric;
+	struct tcpx_pep *tcpx_pep = container_of(pep, struct tcpx_pep,
+						 util_pep.pep_fid);
+
+	tcpx_fabric = container_of(tcpx_pep->util_pep.fabric,
+				   struct tcpx_fabric, util_fabric);
+
+	tcpx_handle = calloc(1, sizeof(*tcpx_handle));
```

Comment:
We should avoid memory allocations in the reject case.  Move any necessary memory allocations to the reporting of the connection.